### PR TITLE
Brandon mentors filter

### PR DESCRIPTION
**Pull request by 54skyxenon**
Branched off of Felicia's mentor modal branch and started working on the filter components. As of right now the filters process data correctly, but don't actually work and are ugly. Will continue fixing over the next week.

### CODE REVIEW

**Comment by 54skyxenon** [comment]
I got filtering by attribute to work (I think), please let me know if it works for you guys! Just need to style the components now. Also, just a heads up: something I noticed is that for one or two mentor objects, clicking the modal breaks the frontend because the `.expertise` key is null.

**Comment by 54skyxenon** [comment]
Got filtering to work by AND'ing all three filter selections together, and also implemented styling in CSS with help from Sarah.

Screen Shot 2021-02-07 at 9 22 49 PM

**Review by snwessel — APPROVED** [review]
@54skyxenon would you mind deleting the `package-lock.json` file and re-running `npm install` if you haven't already tried that? It's weird that the package-lock.json file is still showing changes. 

Other than that, this looks great!

File: src/styles/mentors.scss [hunk 1]
```diff
@@ -133,3 +133,13 @@
...
 }
+
+.mentor-filter {
+  border-radius: 10px;
+  border: none;
+  width: auto;
+  height: 60px;
+  margin: 1em;
+  background-color: $teal-blue;
+  color: white;
```
Comment (snwessel) [inline src/styles/mentors.scss:144]: Oh - one more quick question: do you know why the text is black on one dropdown in the screenshot? I tried to recreate it myself to see what was up but they all look fine in my browser.

Comment (54skyxenon) [reply src/styles/mentors.scss:144]: No, unfortunately not :(

Comment (54skyxenon) [reply src/styles/mentors.scss:144]: I don't think I can actually reproduce it either

**Comment by 54skyxenon** [comment]
@snwessel If I remove `node_modules` and `package-lock.json` and then redo `npm install`, nothing shows up as a new change

**Comment by snwessel** [comment]
@54skyxenon I looked into the `package-lock.json` thing a bit more and I think it's fine. My computer does the same thing when I run `npm-install` locally (even on the main branch). 

I poked around old branches and I think that our `lockfileversion` had been bumped to version 2 somehow and that restructured the whole `package-lock.json` file - and the changes slipped through our code reviews unnoticed. When you run `npm install` now the file goes back to version 1, which is how it probably should be. 

TLDR; It's fine to keep that change in your PR. I'll try to get one more person to review and then we can merge it 😀

File: src/components/Mentors.js [hunk 2]
```diff
@@ -100,18 +151,38 @@ export const Mentors = () => {
...
+          {
+            companies.map(c => (
+              <option value={c}>{c}</option>
+            ))
+          }
+        </select>
+      </Row>
+      <Row>
+        {isLoading && <Spinner animation='border' variant='primary' /> /* TODO import custom styled spinner component*/} 
+        {!data.length && <div>Nothing to see here...</div> /* TODO handle empty array in case of API error (or if its actually empty for any reason)*/}
```
Comment (daniaalnahdi) [inline src/components/Mentors.js]: this condition should be `!data.length && !isLoading`

**Review by daniaalnahdi — APPROVED** [review]
left one small comment, otherwise all good!
